pkg/api/txt.go

```diff
@@ -87,120 +104,140 @@ type TXTBiosMLEFlags struct {
 	IsClientState   bool
 }
 
-func ReadTXTRegs() (TXTRegisterSpace, error) {
+func FetchTXTRegs() ([]byte, error) {
+	data := make([]byte, 1024)
+	if err := ReadPhysBuf(txtPublicSpace, data); err != nil {
+		return nil, err
+	}
+	return data, nil
+}
+
+func ParseTXTRegs(data []byte) (TXTRegisterSpace, error) {
 	var regSpace TXTRegisterSpace
 	var err error
-	var u8 Uint8
-	var u32 Uint32
 	var u64 Uint64
 
-	regSpace.Sts, err = readTXTStatus()
+	regSpace.Sts, err = readTXTStatus(data)
 	if err != nil {
 		return regSpace, err
 
 	}
 
-	regSpace.ErrorCode, regSpace.ErrorCodeRaw, err = readTXTErrorCode()
+	regSpace.ErrorCode, regSpace.ErrorCodeRaw, err = readTXTErrorCode(data)
 	if err != nil {
 		return regSpace, err
 
 	}
 
-	regSpace.Dpr, err = readDMAProtectedRange()
+	regSpace.Dpr, err = readDMAProtectedRange(data)
 	if err != nil {
 		return regSpace, err
 
 	}
 
 	// TXT.ESTS (0x8)
-	err = ReadPhys(txtPublicSpace+0x8, &u8)
+	buf := bytes.NewReader(data[txtEsts:])
+	err = binary.Read(buf, binary.LittleEndian, &regSpace.TxtReset)
+
```

Comment:
Instead of recreating readers, create one for the whole slice and use Seek to move the read pointer.